pygmt/xarray_backend.py

```diff
@@ -0,0 +1,72 @@
+"""
+An xarray backend for reading raster grid/image files using the 'gmt' engine.
+"""
+
+import os
+from pathlib import Path
+from typing import Literal
+
+import xarray as xr
+from pygmt.clib import Session
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import build_arg_list
+from pygmt.src.which import which
+from xarray.backends import BackendEntrypoint
+
+
+class GMTBackendEntrypoint(BackendEntrypoint):
+    """
+    Xarray backend to read raster grid/image files using 'gmt' engine.
+
+    Relies on the libgdal-netcdf driver used by GMT C for NetCDF files, and libgdal for
+    GeoTIFF files.
```

Comment:
This is not 100% dorrect. GMT uses the netCDF libray for reading/writing .nc files, not the libgdal-netcdf driver. I'm even unsure if libgdal-netcdf is needed.

Reply:
Hmm, you're right. I was getting some error when I had a GMT dev `6.6.0.dev5+b3580a7` version installed without `libgdal-netcdf`, but can't seem to reproduce it anymore. So to update the note here, we say that GMT is calling the NetCDF C library directly somehow?

Reply:
> we say that GMT is calling the NetCDF C library directly somehow?

Yes.

Reply:
Done in c5ebf9f27b8cecbbdb998ffe833df838f3cdd200